### PR TITLE
[QuickInstall] Allow Forum Post installation for both Revenge and Vendetta's Plugins / Themes forums

### PR DESCRIPTION
**Pull request by puhbu**
Firstly, yes, I know my commits are cancerous ~~and so is my code~~

## Allow the user to install themes and plugins by holding down on forum posts on both Vendetta's server and Revenge's server.

Worth noting:

- Holding on a THEME(plugins worked for me) forum post that has the install link hyperlinked will make it so the "install" button doesn't appear, but you can press on the hyperlinked "install" and it'll pop up on your screen to install it.

Things To-do:

- [x] Add Revenge's plugin proxy

 I'm open to any comments. I also dislike my code but there's always room for improvement.

### CODE REVIEW

**Review by riivx — APPROVED** [review]

File: src/def.d.ts [hunk 1]
```diff
@@ -464,6 +464,9 @@ interface VendettaObject {
     DISCORD_SERVER_ID: string;
     PLUGINS_CHANNEL_ID: string;
     THEMES_CHANNEL_ID: string;
+    DISCORD_REVENGE_SERVER_ID: string;
+    PLUGINS_REVENGE_CHANNEL_ID: string;
+    THEMES_REVENGE_CHANNEL_ID: string;
```
Comment (PalmDevs) [inline src/def.d.ts:469]: Don't think we need to put type definitions for the new constants here yet, maybe after #29 is completed we can put our own APIs definitions.

**Review by PalmDevs — APPROVED** [review]
Runtime side looks good

File: src/def.d.ts (hunk 1 above)
Comment (puhbu) [reply src/def.d.ts:469]: Got it, then in that case I'll wait for 29 to be completed and then discuss about it. To be completely honest, I once managed to get it work without defining the constants in def.d.ts, but I don't know what black magic there was.

Comment (PalmDevs) [reply src/def.d.ts:469]: Oh no, #29 isn't blocking this PR. The type definitions provided in the d.ts file is just for generating the types package IIRC. It should work without the new type additions.

**Review by nexpid — CHANGES_REQUESTED** [review]
maybe turn `THEMES_REVENGE_CHANNEL_ID` and `PLUGINS_REVENGE_CHANNEL_ID` into arrays that includes both Vendetta and Revenge channel ids?

**Review by she11sh0cked — CHANGES_REQUESTED** [review]

**Comment by she11sh0cked** [comment]
> maybe turn `THEMES_REVENGE_CHANNEL_ID` and `PLUGINS_REVENGE_CHANNEL_ID` into arrays that includes both Vendetta and Revenge channel ids?

Ah didn't see your review :)

@puhbu If you like I can PR my requested changes to your branch. Got some TS experience anyway from my job ^^

**Pull request closed without merging**